### DISCUSSION TITLE
Can't Install On MacOS Big Sur

### DISCUSSION THREAD

**Discussion by stevetalkowski** [Autodesk/maya-usd#920] [Q&A] [answered]
Hello!

I got the following error when attempting to install the plugin on MacOS Big Sur V 11.0.1 (I just recently updated).

Is there a way to manually install this? I enjoy the convenience of the .dmg file, but am willing to do it manually if it's easy enough. I've successfully installed in on my Windows machine.

Also, will this plug-in allow exporting skeletal animation? I'm primarily interested in getting some textured, animated robot characters out from Maya and over to Apple's Reality Composer.

Thx!

-Steve

![Screen Shot 2020-11-15 at 4 11 14 PM](https://user-images.githubusercontent.com/22301087/99200978-7f881a80-275d-11eb-895c-716ccf34e5eb.png)

**Comment by dgovil** [comment]
If you right click and hit open, it should give you the ability to bypass the quarantine check

It's an issue caused by the dylibs not being notarized, and because you downloaded them, they are quarantined for protection.

**Reply by stevetalkowski** [reply]
Morning Dhruv!

Ok, so I did the right click and got to the installer, but now, i'm experiencing something I've never seen before. My Macintosh HD is not available to select, however, I'm seeing these other partitions that are part of my "Container Disk1". I don't recall seeing this before. Maybe it's a new thing in Big Sur? Any suggestions on how I should go about installing? Would it work if I just selected a folder on my Drobo, then manually move the files to the required folders on the main SSD? Thanks for any help or suggestions!

![Screen Shot 2020-11-16 at 9 52 33 AM](https://user-images.githubusercontent.com/22301087/99289579-01c21e80-27f2-11eb-876a-553511c778e9.png)

**Reply by dgovil** [reply]
Huh, unfortunately, I have no experience with the installer itself. The Autodesk folk might be better suited to answer that one.

**Reply by stevetalkowski** [reply]
Thx. I'll try reaching out to a few of my Maya contacts at Autodesk and let you know once I get it to work. 👍

**Reply by kxl-adsk** [reply]
You don't need to look for contacts...here is the right place to ask. I don't have an answer at hand, but we will get back to you.

**Reply by stevetalkowski** [reply]
> You don't need to look for contacts...here is the right place to ask. I don't have an answer at hand, but we will get back to you.

Ah, thanks Krystian! Appreciate it. :)

**Reply by santosg87** [reply]
I can reproduce this internally as well.

although, i noticed that no partition is selected on that page. 

continuing the installation without selecting anything will place the files on the expected locations. :)

we will dig a bit deeper.

Cheers!

**Reply by stevetalkowski** [reply]
> although, i noticed that no partition is selected on that page.

I didn't know which one to select, since MacintoshHD wasn't available.

> continuing the installation without selecting anything will place the files on the expected locations. :)

I did try to continue without selecting anything, and it failed on my end. I've attached additional screenshots.

> we will dig a bit deeper.

Thx!
![Screen Shot 2020-11-17 at 11 43 44 AM](https://user-images.githubusercontent.com/22301087/99439802-854f3e80-28ca-11eb-834e-b6e6a5229759.png)
![Screen Shot 2020-11-17 at 11 44 03 AM](https://user-images.githubusercontent.com/22301087/99439806-85e7d500-28ca-11eb-8e24-2a9067686791.png)
![Screen Shot 2020-11-17 at 11 44 16 AM](https://user-images.githubusercontent.com/22301087/99439807-86806b80-28ca-11eb-944a-21276d429ebc.png)

**Comment by dgovil** [comment, marked as answer]
So i went spelunking out of curiosity...
If you launch the app you get a different installer than if you do the following:

* Right click on the installer app and select "Show Package Contents"
* Navigate to `Contents/Packages/MayaUSD_core_2020_2020.pkg` and run it.

It gives an installer that doesn't hit the issues above, and looks like it installed correctly for me?
It looks like the Installer App is using Qt to install, so it may be an issue with the Qt installer...

Anyway, worth a try.

**Reply by stevetalkowski** [reply]
SUCCESS! I  had to right click it on it again to bypass the security warning.

I then created a simple cube and exported it. I double clicked the exported file and it opened up in Xcode. I noticed that it has subdivision applied. Went back to Maya and saw that Subdivision Method is set to Catmull-Clark as default. Maybe that should be set to None for expected one-to-one output? Though, i'm glad to see that as an option, as I tend to box model with clean quads.

I then sent the file over to my iPhone 12 Pro Max, wanting to see if I could next read it into Reality Composer. RC is not recognizing the file. Previously, i've been able to load files exported as USDZ from Substance Painter. Is there a difference between USD and USDZ? Is there something I need to flag upon export to get it saved as a USDZ file that can be read into RC?

![Screen Shot 2020-11-17 at 4 57 41 PM](https://user-images.githubusercontent.com/22301087/99470343-6fa53d80-28f9-11eb-86b5-9e80a18196c3.png)

**Reply by dgovil** [reply]
Reality Composer only supports usdz files.

## Differences between usd and usdz
`.usd` files default to being `.usdc` or can be `.usda` if specified. However they can't be `.usdz` afaik.

`.usdz` is basically a specially zipped version of USD files and their dependencies. Sort of like if your entire usd+dependencies was a folder and then zipped to be a single file. 

Note you cannot zip up a usd to make a usdz, there is some special sauce to it (alignment, ordering, no compression etc) but you can use usdzip, usdzconverter or Reality Converter to create one from usd files or other compatible formats.

##  Creating a usdz from Maya

If you specify the `.usdz` extension when writing out a file from Maya, it'll create a proper usdz file for you.
Alternatively, you can use [Reality Converter ](https://developer.apple.com/news/?id=01132020a) to convert it for you.

**Reply by santosg87** [reply]
to export usdz, you should be able to by simply adding the file format  manually on maya when typing in the file name.
(i.e. "yourCube.usdz"instead of just yourCube.),

maya would then take care of the rest when exporting. :)

**Reply by stevetalkowski** [reply]
Hey Dhruv, thanks, that works. Seems a bit counter-intuitive to have to remember appending the Z though. Shouldn't that be a check mark in the export dialogue window perhaps?

I've been busy, but just tried another example. A simple elongated, segmented and textured cube that is bound to a two joint skeleton with animation. I selected the skeleton and geometry and exported with the .USDZ extension and check marked the animation option. When I double click that file, it opens up Xcode. The associated texture is missing, and there is no animation. I then tried selecting the two joints that have simple rotation and baked those keys (48 frame sequence) and tried re-exporting. Still no textures or animation. Do you have any guidance on the preferred method to get this to work out from Maya?

**Reply by dgovil** [reply]
AFAIK there was a bug in one of the earlier commits that caused textures to not be embedded in the usdz. I don't think the fix landed in time for the 0.5 release (https://github.com/Autodesk/maya-usd/pull/865#event-3922349296)

For the skeleton, the Export Skeleton option of the usd exporter is off by default and I don't believe the option is exposed in the UI right now. You can export via MEL/Python calls to mayaUSDExport to get the skeleton in.

I have a PR we're preparing to add Blendshape export that should also add back the skeleton export options if accepted. Note: Blendshapes themselves aren't supported in RealityKit yet but skeletons are.

```
cmds.mayaUSDExport(file=usdFile, exportSkels='auto')
```

----

Unfortunately, for the textures issue you'll likely need to build this repo yourself or wait till a 0.6 release is cut.
For the skeleton, you'll need to use the python export command to do it at the moment, till the skeleton options are exposed in the UI.

**Reply by stevetalkowski** [reply]
*sigh* Ok. I'm just looking for a quick, down N dirty pipeline to get a textured and animated character out from Maya and into a useable format for Reality Composer. Seems like it's not "Ready for Prime Time" at the moment.

When you suggest solution like "export via MEL/Python calls" and "build the repo yourself", my non-programmer artist brain starts to glaze over! ;)

I'm on the PR, but haven't been active that much with it. When I get some more time, i'll take a look at getting it up and running on both the Mac and PC.

**Comment by kxl-adsk** [comment]
Update
We haven't addressed the reported issue in the 0.6.0 installer, but the work is in progress and should be ready for the next drop.
